src/SearchAction.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace strongsquirrel\actions;
+
+use yii\base\Action;
+use yii\base\InvalidConfigException;
+use yii\base\Model;
+
+/**
+ * Class SearchAction
+ *
+ * @package strongsquirrel\actions
+ */
+class SearchAction extends Action
+{
+    const HTTP_METHOD_GET = 'get';
+    const HTTP_METHOD_POST = 'post';
+
+    /**
+     * @var string
+     */
+    public $scenario = Model::SCENARIO_DEFAULT;
+
+    /**
+     * @var string the name of the view action.
+     */
+    public $view = 'search';
+
+    /**
+     * @var string
+     */
+    public $httpMethod = self::HTTP_METHOD_POST;
```

Comment:
Then we need to write about source for model. To rename the variable would be better option. And replace ```$data = call_user_func([\Yii::$app->request, $this->httpMethod]);``` to simple if/else.